docs/proposals/20210222-machineadm.md

```diff
@@ -0,0 +1,835 @@
+</td>
+<td>Provisioning</td><td>U7</td>
+</tr>
+
+<tr>
+<td>R12</td>
+<td>
+When the machine bootstrapper is downloaded on boot the location MUST be configurable</td>
+<td>Preparation</td><td>U8</td>
+</tr>
+
+<tr>
+<td>R13</td>
+<td>
+When the machine bootstrapper is downloaded from the public internet, it MUST be downloadable from a location not subject to frequent rate limiting (e.g. a GCS bucket).</td>
+<td>Preparation</td><td>U9</td>
+</tr>
+
+<tr>
+<td>R14</td>
+<td>
+The machine bootstrapper MUST be able to configure containerd given a structured configuration input..</td>
+<td>Provisioning</td><td>U10</td>
+</tr>
+
+<tr>
+<td>R15</td>
+<td>
+The machine bootstrapper MUST publish a documented contract for operating system maintainers to integrate with the machine bootstrapper.
+</td>
+<td>All</td><td>U1</td>
+</tr>
+
+
+</tbody>
+
+
+</table>
+
+
+
+
+### Implementation details
+
+#### Deployment Model
+
+<table>
+<thead>
+<tr><th>Component</th><th>Location</th><th>Description</th>
+</thead>
+<tbody>
+
+<tr>
+<td>machineadm-bootstrap-controller</td><td>cluster-api repo</td>
+<td>
+The core machine bootstrapper contains the controllerst hat reconciles MachineConfig to produce machineadm yaml
+</td>
+
+<tr>
+<td>machineadm-bootstrap-controller</td><td>cluster-api repo</td>
+<td>
+The core machine bootstrapper contains the controllerst hat reconciles MachineConfig to produce machineadm yaml
+</td>
+</tr>
+
+<tr>
+<td>machineadm cli</td><td>cluster-api repo</td>
+<td>
+The core CLI that reads machineadm.yaml and orchestrates the bootstrapping workflow
+</td>
+</tr>
+
+<tr>
+<td>machineadm core plugins</td><td>cluster-api repo</td>
+<td>
+A controller that ships with the infrastructure provider that can reconcile the storage of machineadm configurations with infrastructure APIs (e.g. Amazon S3/Minio, GCS, custom server etc...).
+</td>
+</tr>
+
+<tr>
+<td>machineadm remote configuration plugin</td><td>infrastructure provider repo</td>
+<td>
+An infrastructure plugin for the node bootstrapper will be created within infrastructure providers that would be responsible for infrastructure-specific logic to pull files machineadm files from a given location and upload
+bootstrap data.
+</td>
+</tr>
+
+<tr>
+<td>machineadm encryption plugin</td><td>infrastructure provider repo</td>
+<td>
+An infrastructure plugin for the node bootstrapper will be created within infrastructure providers that would be responsible for infrastructure-specific logic to use an external key management service.
+</td>
+</tr>
+
+<tr>
+<td>machineadm infrastructure controller</td><td>infrastructure provider repo</td>
+<td>
+A controller that ships with the infrastructure provider that can reconcile the storage of machineadm configurations with infrastructure APIs (e.g. Amazon S3/Minio, GCS, custom server etc...).
+</td>
+</tr>
+
+</tbody>
+</table>
+
+##### Core plugins
```

Comment:
How was this selection made? What would be a criteria for a core plugin?

Reply:
I don't have a good metric for this right now. Core plugins should do at least what we expose in cloud-init across the providers today in order to enable upgrades.